File: src/pt/joaomneto/titancompanion/util/DiceRoller.java
package pt.joaomneto.titancompanion.util;

import java.util.Random;

import pt.joaomneto.titancompanion.adventure.impl.util.DiceRoll;

public abstract class DiceRoller {


	private static Random random = new Random(System.currentTimeMillis()); 
	
	public static int rollD6(){
		return random.nextInt(6)+1;
	}

	public static DiceRoll roll2D6(){
		return new DiceRoll(rollD6(),rollD6());
	}
	

	public static int roll3D6(){
		return rollD6()+rollD6()+rollD6();
	}
	
}

File: src/main/java/pt/joaomneto/titancompanion/util/DiceRoller.kt
package pt.joaomneto.titancompanion.util

import pt.joaomneto.titancompanion.adventure.impl.util.DiceRoll

import java.util.Random

object DiceRoller {

    private val random = Random(System.currentTimeMillis())

    fun rollD6(): Int {
        return random.nextInt(6) + 1
    }

    fun roll2D6(): DiceRoll {
        return DiceRoll(rollD6(), rollD6())
    }

    fun roll3D6(): Int {
        return rollD6() + rollD6() + rollD6()
    }
}


